src/context/product-page-context.js

```diff
@@ -1,23 +1,43 @@
-import { createContext, useContext, useReducer } from "react";
+import { createContext, useContext, useReducer, useEffect } from "react";
+import { apiCall } from "../utility/apiCall";
 
 const ProductContext = createContext();
 const useProductContext = () => useContext(ProductContext);
+export const encodedToken = localStorage.getItem("token");
 
 const productPageReducer = (productPageState, action) => {
-  switch(action.type) {
-    case 'CATEGORY_NAME':
-      return {...productPageState, categoryName: action.payload}
+  switch (action.type) {
+    case "CATEGORY_NAME":
+      return { ...productPageState, categoryName: action.payload };
+    case "LIST":
+      return { ...productPageState, productList: action.payload };
```

Comment:
suggestion: you can use better naming here. Instead of case "LIST", case "SET_PRODUCT_LIST" would make it more clear.